ple/games/donkeykong/Board.py

```diff
@@ -0,0 +1,335 @@
+__author__ = 'Batchu Vishal'
+import pygame
+import math
+import sys
+
+from Person import Person
+from OnBoard import OnBoard
+from Coin import Coin
+from Player import Player
+from Fireball import Fireball
+from Donkey import Donkey
+
+'''
+This class defines our gameboard.
+A gameboard contains everthing related to our game on it like our characters, walls, ladders, coins etc
+The generation of the level also happens in this class.
+'''
+
+
+class Board:
+    def __init__(self, width, height, rewards, rng):
+        self.__width = width
+        self.__actHeight = height
+        self.__height = self.__actHeight + 10
+        self.score = 0
+        self.rng = rng
+        self.rewards = rewards
+        self.gameState = 1
+        self.cycles = 0  # For the characters animation
+        self.direction = 0
+
+        self.white = (255, 255, 255)
+
+        '''
+        The map is essentially an array of 30x80 in which we store what each block on our map is.
+        1 represents a wall, 2 for a ladder and 3 for a coin.
+        '''
+        self.map = []
+        # These are the arrays in which we store our instances of different classes
+        self.Players = self.Enemies = self.Allies = self.Coins = self.Walls = self.Ladders = self.Fireballs = self.Hearts = self.Boards = self.FireballEndpoints = []
+
+        # Resets the above groups and initializes the game for us
+        self.resetGroups()
+
+        self.myfont = pygame.font.SysFont("comicsansms", 50)
+
+        self.background = pygame.image.load('Assets/background.png')
+        self.background = pygame.transform.scale(self.background, (width, height))
+
+        # Initialize the instance groups which we use to display our instances on the screen
+        self.fireballGroup = pygame.sprite.RenderPlain(self.Fireballs)
+        self.playerGroup = pygame.sprite.RenderPlain(self.Players)
+        self.enemyGroup = pygame.sprite.RenderPlain(self.Enemies)
+        self.wallGroup = pygame.sprite.RenderPlain(self.Walls)
+        self.ladderGroup = pygame.sprite.RenderPlain(self.Ladders)
+        self.coinGroup = pygame.sprite.RenderPlain(self.Coins)
+        self.allyGroup = pygame.sprite.RenderPlain(self.Allies)
+        self.fireballEndpointsGroup = pygame.sprite.RenderPlain(self.FireballEndpoints)
+        self.boardGroup = pygame.sprite.RenderPlain(self.Boards)
+        self.heartGroup = pygame.sprite.RenderPlain(self.Hearts)
+
+    def resetGroups(self):
+        self.score = 0
+        self.map = []  # We will create the map again when we reset the game
+        self.Players = [Player(pygame.image.load('Assets/still.png'), (50, 440), self.rng)]
```

Comment:
performance: might be useful to use image.convert() or image.convert_alpha() depending on if the image uses alpha